.github/workflows/java.yaml

```diff
@@ -0,0 +1,36 @@
+name: Maven Test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # 4.1.7
+
+      - name: Set up JDK
+        uses: actions/setup-java@2dfa2011c5b2a0f1489bf9e433881c92c1631f88 # 4.3.0
+        with:
+          java-version: "21"
+          distribution: "temurin"
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+        with:
+          path: |
+            ~/.m2/repository
+            target
+          key: maven-${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-${{ runner.os }}-
+
+      - name: Make mvnw executable
+        run: chmod +x ./mvnw
+
+      - name: Run tests
+        run: mvn test
```

Comment:
I stedet for at gøre filen eksekverbar hver eneste gang din github action kører, kan du gøre den eksekvérbar en enkelt gang og commite det til GitHub:

```
chmod +x ./mvnw
git add ./mvnw
git commit -m "fix: chmod +x ./mvnw"
```

Men det virker ikke nødvendigt at gøre den eksekverbar hvis du i stedet skriver `mvn test`. Filen `mvnw` er en wrapper til at køre Maven på systemer hvor det ikke er sat ordentligt op. Men hvis din `mvn`-kommando virker fint, er `./mvnw` måske mindre vigtig.

En sidste bemærkning: Ligesom du har `mvn test` kan du også lave et trin før det, hvor man skriver `mvn build`. Hvis du gør det, kan man bedre skelne i CI om noget er en byggefejl eller en test der fejler. Ellers vil den fejle på testtrinnet når der er byggefejl, og det kan tage sekunder eller minutter at lure, hvorimod hvis den fejler med overskriften "Build", så ved man at det er en Java-fejl og ikke en logisk fejl i en test.

Ellers ser det godt ud!